Source/WTF/wtf/WeakHashSet.h

```diff
@@ -195,15 +195,30 @@ class WeakHashSet final {
 
     void forEach(NOESCAPE const Function<void(T&)>& callback)
     {
-        auto items = map(m_set, [](const Ref<WeakPtrImpl>& item) {
-            auto* pointer = static_cast<T*>(item->template get<T>());
-            return WeakPtr<T, WeakPtrImpl> { pointer };
-        });
-        for (auto& item : items) {
-            // FIXME: This contains check is only necessary if the set is being mutated during iteration.
-            // Change it to an assertion, or make this function use begin() and end().
-            if (item && m_set.contains(*item.m_impl))
-                callback(*item);
+        if constexpr (WTF::HasRefPtrMemberFunctions<T>::value) {
+            auto items = compactMap(m_set, [](const Ref<WeakPtrImpl>& item) -> RefPtr<T> {
+                return RefPtr { static_cast<T*>(item->template get<T>()) };
+            });
+            for (auto& item : items)
+                callback(item.get());
+        } else if constexpr (WTF::HasCheckedPtrMemberFunctions<T>::value) {
+            auto items = compactMap(m_set, [](const Ref<WeakPtrImpl>& item) -> CheckedPtr<T> {
+                return CheckedPtr { static_cast<T*>(item->template get<T>()) };
+            });
+            for (auto& item : items)
+                callback(item.get());
+        } else {
+            // Exception case: unsafe.
```

Comment:
In a follow-up, I'll prioritize getting rid of this code path. This is a good way to prioritize which types to drop the WeakPtr refcounting/checkedptr exception for.